File: core/SyncPlatform/src/Services/Notification/Notification.Domain/Models/NotificationTemplate.cs
using MongoDB.Bson.Serialization.Attributes;
using Notification.Domain.Models;
using Notification.Domain.Enums;

namespace Notification.Domain.Models;

public class NotificationTemplate : BaseMongoEntity
{
    public string TemplateCode { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string DefaultTitle { get; set; } = string.Empty;

    public string DefaultBody { get; set; } = string.Empty;

    [BsonIgnoreIfNull]
    public string? VariablesJson { get; set; }

    public NotificationChannel Channel { get; set; }

    public bool IsActive { get; set; }
}

File: ui/web/src/lib/validations/notification-template.ts
import { z } from "zod";

export const notificationTemplateSchema = z.object({
  templateCode: z.string().min(1, "Required").max(128),
  name:         z.string().min(1, "Required").max(256),
  defaultTitle: z.string().min(1, "Required").max(256),
  defaultBody:  z.string().min(1, "Required").max(2000),
  variablesJson: z.string().optional(),
  channel:      z.string().min(1, "Required"),
  isActive:     z.boolean().default(true),
});

export type NotificationTemplateFormValues = z.infer<typeof notificationTemplateSchema>;
